File: src/src_common/fmg/common/ui/ITimer.java
package fmg.common.ui;

public interface ITimer extends AutoCloseable {

   /** milliseconds */
   long getInterval();
   /** milliseconds */
   void setInterval(long delay);

   /** set null - stop timer; otherwise - started */
   void setCallback(Runnable cb);

   @Override
   void close();

}

File: src/src_common/fmg/common/ui/ITimer.cs
using System;

namespace fmg.common.ui {

   public interface ITimer : IDisposable {

      /// <summary> in miliseconds </summary>
      long Interval { get; set; }

      /// <summary> set null - stop timer; otherwise - started </summary>
      Action Callback { set; }

   }

}
